lua/telekasten.lua

```diff
@@ -371,45 +374,55 @@ local function imgFromClipboard()
             return
         end
 
-        local get_paste_command
-        if vim.fn.executable("xsel") == 1 then
-            get_paste_command = function(dir, filename)
-                local _image_path = vim.fn.system("xsel --clipboard --output")
-                local image_path = _image_path:gsub("file://", "")
-                if
-                    vim.fn
-                        .system("file --mime-type -b " .. image_path)
-                        :gsub("%s+", "")
-                    == "image/png"
-                then
-                    return "cp " .. image_path .. " " .. dir .. "/" .. filename
-                else
-                    return ""
-                end
+        local paste_command = {}
+        paste_command["xsel"] = function(dir, filename)
+            local _image_path = vim.fn.system("xsel --clipboard --output")
+            local image_path = _image_path:gsub("file://", "")
+            if
+                vim.fn
+                    .system("file --mime-type -b " .. image_path)
+                    :gsub("%s+", "")
+                == "image/png"
+            then
+                return "cp " .. image_path .. " " .. dir .. "/" .. filename
+            else
+                return ""
             end
+        end
+        paste_command["xclip"] = function(dir, filename)
+            return "xclip -selection clipboard -t image/png -o > "
+                .. dir
+                .. "/"
+                .. filename
+        end
+        paste_command["osascript"] = function(dir, filename)
+            return string.format(
+                'osascript -e "tell application \\"System Events\\" to write (the clipboard as «class PNGf») to '
+                    .. '(make new file at folder \\"%s\\" with properties {name:\\"%s\\"})"',
+                dir,
+                filename
+            )
+        end
+
+        paste_command["wl-paste"] = function(dir, filename)
+            return "wl-paste -n -t image/png > " .. dir .. "/" .. filename
+        end
+
+        local get_paste_command
+        if paste_command[M.Cfg.clipboard_program] ~= nil then
+            get_paste_command = paste_command[M.Cfg.clipboard_program]
```

Comment:
Wouldn't it be better to first check if the selected program is indeed installed, rather than returning directly a command that may fail? 

Something like
``` if paste_command[M.Cfg.clipboard_program] ~= nil and vim.fn.executable(M.Cfg.clipboard_program) == 1```

I can already tell that someone is going to eventually write something that is not supported, or make a typo in the name of the program. We should catch that.

Reply:
Tackled in 6bc2dc6c0cabc885b4e494201bf4810f0142530f. I added an extra if with a meaningful message.

Reply:
Awesome